docs/reference/docs/data-replication.asciidoc

```diff
@@ -0,0 +1,136 @@
+that the problematic shard be removed from the in-sync replica set. Only once this has successfully been done, does
+the primary acknowledge the operation. Note that the master will also instruct another node to start building a new
+shard copy in order to restore the system to a healthy state.
+
+While forwarding an operation to the replicas, the primary will use the replicas to validate that it is still the
+active primary. If the primary has been isolated due to a network partition (or a long GC) it may continue to process
+ incoming indexing operations before realising that it has been demoted. Operations that come from a stale primary
+ will be rejected by the replicas. The primary, following its protocol, will then reach out to the master and will
+ learn that it has been replaced. The operation is then routed to the new primary.
+
+What happens if there are no replicas? This is a valid scenario that can happen due to index configuration or simply
+because the replicas have failed. In that case the primary is processing operations without any external validation,
+which may seem problematic. On the other hand, the primary cannot fail other shards on its own. This means that the
+master knows that the primary is the only single good copy. We are therefore guaranteed that the master will not promote
+any other (out-of-date) shard copy to be a  new primary and that any operation indexed into the primary will not be lost.
+Of course, since at that point we run with only single copy of the data, physical hardware issues can cause data loss
+(see <<index-wait-for-active-shards>> for some mitigation options).
+
+=== Basic read model
+
+Reads in Elasticsearch can be very lightweight lookups by ID or a heavy search request with complex aggregations that
+take non-trivial CPU power. One of the beauties of the primary-backup model is that it keeps all shard copies identical
+(with the exception of in-flight operations).. As such, a single copy is sufficient to serve a read requests.
+
+When a read request is sent to a node, that node is responsible for forwarding it to the right places, collating the
+responses, and responding to the client. We call that node the coordinating node for that search. That node does so
+according to this basic flow:
+
+. Resolve the read requests to the relevant shards. Note that since most searches will be sent to one index or more,
+   they typically need to read from multiple shards, each representing a different subset of the data.
+. For each relevant shard, select an active copy from the shard replication group. This can be either the primary or
+   a replica. By default, Elasticsearch will simply round robin between the shard copies.
+. Send shard level read requests to the selected copies.
+. Combine the results and respond. Note that in the case of get by ID look up, only one shard is relevant and this step can be skipped.
```

Comment:
And in the case of search there is usually another phase to fetch fields from the top results.